include/host/wasi/environ.h

```diff
@@ -38,6 +43,57 @@ class Environ {
 
   void fini() noexcept;
 
+  WasiExpect<void> getAddrInfo(const char *Node, const char *Service,
+                               const __wasi_addrinfo_t *Hint,
+                               /*Out*/ addrinfo **Res,
+                               __wasi_size_t *ResLength) {
+    struct addrinfo TmpHint;
+    struct addrinfo *TmpResult = NULL;
+    struct addrinfo *TmpPointer;
+    int POSIXReturn;
+
+    *ResLength = 0;
+    TmpHint.ai_flags = Hint->ai_flags;
+    TmpHint.ai_family = Hint->ai_family;
+    TmpHint.ai_socktype = Hint->ai_socktype;
+    TmpHint.ai_protocol = Hint->ai_protocol;
+    TmpHint.ai_addrlen = Hint->ai_addrlen;
+    TmpHint.ai_addr = NULL;
+    TmpHint.ai_canonname = NULL;
+    TmpHint.ai_next = NULL;
+
+    POSIXReturn = ::getaddrinfo(Node, Service, &TmpHint, &TmpResult);
+    switch (POSIXReturn) {
+    case EAI_ADDRFAMILY:
+      return WasiUnexpect(__WASI_ERRNO_AIADDRFAMILY);
+    case EAI_AGAIN:
+      return WasiUnexpect(__WASI_ERRNO_AIAGAIN);
+    case EAI_BADFLAGS:
+      return WasiUnexpect(__WASI_ERRNO_AIBADFLAG);
+    case EAI_FAIL:
+      return WasiUnexpect(__WASI_ERRNO_AIFAIL);
+    case EAI_FAMILY:
+      return WasiUnexpect(__WASI_ERRNO_AIFAMILY);
+    case EAI_MEMORY:
+      return WasiUnexpect(__WASI_ERRNO_AIMEMORY);
+    case EAI_NODATA:
+      return WasiUnexpect(__WASI_ERRNO_AINODATA);
+    case EAI_NONAME:
+      return WasiUnexpect(__WASI_ERRNO_AINONAME);
+    case EAI_SERVICE:
+      return WasiUnexpect(__WASI_ERRNO_AISERVICE);
+    case EAI_SOCKTYPE:
+      return WasiUnexpect(__WASI_ERRNO_AISOCKTYPE);
+    case EAI_SYSTEM:
+      return WasiUnexpect(__WASI_ERRNO_AISYSTEM);
+    }
+    for (TmpPointer = TmpResult; TmpPointer != nullptr;
+         TmpPointer = TmpPointer->ai_next) {
+      (*ResLength)++;
```

Comment:
Looks like you only calculate the size of linked list. But I didn't find anything related to covert this linked list to a linear array.

Reply:
Yes, I covert this linked list to a linear array in`wasifunc.cpp` ,not in `environ.h`. this function in `environ.h` just output an linked list and the size of linked list

Reply:
Is this correct?